src/WebJobs.Extensions.DurableTask/LinuxLogger.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    public class LinuxLogger : ILogger
+    {
+        private readonly bool inConsumption;
+        private readonly string consolePrefix = "MS_DURABLE_FUNCTION_EVENTS_LOGS";
+        private readonly string loggingPath = "/var/log/functionsLogs/durableevents.log";
+
+        public LinuxLogger(bool inConsumption)
+        {
+            this.inConsumption = inConsumption;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            // TODO: I this we can leave this unimplemented
+            throw new NotImplementedException();
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            // TODO: assuming these logs should never be disabled
+            return true;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            string jsonString = ""; // assume that, somehow, this becomes a string-representation of a JSON-representation of an ETW event....
+            // @Chris, any ideas on how to set the right value here? I suppose we'll need to be listening to ETW events as well and transform them into a JSON-string here.
+            if (this.inConsumption)
+            {
+                // In Linux Consumption, we write to console
+                string consoleLine = this.consolePrefix + " " + jsonString;
+                Console.WriteLine(consoleLine);
+            }
+            else
+            {
+                // In Linux Dedicated, we write to file
+                using (var streamWriter = new StreamWriter(this.loggingPath, true))
+                {
+                    streamWriter.WriteLine(jsonString);
```

Comment:
Synchronous writes to files worry me a little. We might need to use `WriteLineAsync` instead to avoid blocking I/O. However, the crazy side of me wants to do it in a way that is _completely_ asynchronous - i.e. don't even await. Something like:
```csharp
Task unused = streamWriter.WriteLineAsync(jsonString);
```

Though I have some questions about whether this will cause problems if there is an exception writing to a file. I would need to consult with the latest .NET literature to understand how the CLR handles exceptions for unobserved tasks (they used to crash the process, but I think that policy was changed in more recent versions of .NET).

Reply:
Good point!

Let's say we were to go for the more conservative approach, so `await writer.WriteLineAsync(jsonString);`.
If so, we'd have to add the `async` keyword to the method declaration, right? Not sure how this works in .NET, but I'd assume this would then deviate from the ILogger required method names, so I'm not sure if it would get called correctly.

Reply:
That's true. I think logging in general is supposed to have a minimal impact on runtime performance, which is why they did not design this API to by async-compatible, so my original proposal is probably the way to go.

Reply:
Sounds good. I went with the unobserved Task approach, assuming that was your original proposal.

RE: Will Unobserved exceptions crash the process?
The **remarks** section here would suggest not: https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskscheduler.unobservedtaskexception?view=netcore-3.1

It also seems to apply to .NET Standard 2.0, which I think is the one we're on, as  our `.csproj` file